server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala

```diff
@@ -1,167 +1,117 @@
 package uk.gov.ons.addressIndex.server.modules
 
 import javax.inject.{Inject, Singleton}
-
 import uk.gov.ons.addressIndex.server.model.dao.ElasticClientProvider
 import com.google.inject.ImplementedBy
 import com.sksamuel.elastic4s.ElasticDsl._
 import com.sksamuel.elastic4s._
 import play.api.Logger
-import uk.gov.ons.addressIndex.crfscala.CrfScala.{CrfTokenResult, Input}
-import uk.gov.ons.addressIndex.model.db.index.{NationalAddressGazetteerAddress, NationalAddressGazetteerAddresses, PostcodeAddressFileAddress, PostcodeAddressFileAddresses}
-import uk.gov.ons.addressIndex.parsers.Tokens
-import uk.gov.ons.addressIndex.parsers.Tokens.Token
-
+import uk.gov.ons.addressIndex.crfscala.CrfScala.CrfTokenResult
+import uk.gov.ons.addressIndex.model.{AddressScheme, BritishStandard7666, PostcodeAddressFile}
+import uk.gov.ons.addressIndex.model.db.index.HybridIndex
+import uk.gov.ons.addressIndex.server.modules.Model._
 import scala.concurrent.{ExecutionContext, Future}
 
+object Model {
+  case class Pagination(offset: Int, limit: Int)
+  implicit class AutoPaginate(searchDefinition: SearchDefinition) {
+    def paginate(implicit p: Pagination): SearchDefinition = {
+      searchDefinition start p.offset limit p.limit
+    }
+  }
+
+  implicit class AutoSource(searchDefinition: SearchDefinition) {
+    def format(implicit optFmt: Option[AddressScheme]) = {
+      optFmt map {
+        case _: PostcodeAddressFile => searchDefinition sourceExclude HybridIndex.Fields.lpi
+        case _: BritishStandard7666 => searchDefinition sourceExclude HybridIndex.Fields.paf
+      } getOrElse searchDefinition
+    }
+  }
+}
+
 @ImplementedBy(classOf[AddressIndexRepository])
-trait ElasticsearchRepository {
+trait ElasticSearchRepository {
+
   /**
     * An ElasticClient.
     */
-  def client() : ElasticClient
+  def client: ElasticClient
 
-  /**
-    * Query the PAF addres index by UPRN.
-    *
-    * @param uprn the identificator of the address
-    * @return Fucture containing a PAF address or `None` if not in the index
-    */
-  def queryPafUprn(uprn : String) : Future[Option[PostcodeAddressFileAddress]]
+  def logger: Logger
 
   /**
-    * Query the NAG addres index by UPRN.
     *
-    * @param uprn the identificator of the address
-    * @return Fucture containing a NAG address or `None` if not in the index
+    * @param tokens
+    * @param p
+    * @param fmt
+    * @return
     */
-  def queryNagUprn(uprn : String) : Future[Option[NationalAddressGazetteerAddress]]
+  def queryAddress(tokens: Seq[CrfTokenResult])
+    (implicit p: Pagination, fmt: Option[AddressScheme]): Future[RichSearchResponse] = {
```

Comment:
The underlying usage required implicit params, I could move them to the underlying usage.
The idea is that you don't have to worry about the format and pagination, let the scala compiler do the magic :)